packages/admin/src/Charcoal/Admin/ServiceProvider/AdminServiceProvider.php

```diff
@@ -203,6 +206,31 @@ protected function registerAdminServices(Container $container)
         });
     }
 
+    /**
+     * @param Container $container A Pimple DI container.
+     * @return void
+     */
+    protected function registerModelDependencies(Container $container)
+    {
+        /**
+         * @param Container $container A Pimple DI container.
+         * @return array The model dependencies array.
+         */
+        $container->extend('model/dependencies', function (array $modelDependencies, Container $container) {
+            $modelDependencies['view'] = $container['admin/view'];
+            return $modelDependencies;
+        });
+    }
+
+    /**
+     * @param  Container $container A service container.
+     * @return void
+     */
+    protected function registerRouteServices(Container $container)
+    {
+        $container['route/controller/template/class'] = AdminTemplateRoute::class;
```

Comment:
Similar to my comment on `model/dependencies`, I think we need some kind of alternative means to assign a default route controller.

In some projects, such as [Station](https://bitbucket.org/locomotivemtl/station-api/src/bba8b912722b9628649f5dd4488c0dd27d2d7394/config/admin.php#lines-37:46), the custom default route controller is injected into all the routes:

```php
if (!($this['admin'] instanceof AdminConfig)) {
    $this['admin'] = new AdminConfig($this['admin']);
}

// …

$routes = $this['admin']['routes'];
foreach ($routes['templates'] as $routeIdent => &$routeStruct) {
    if (empty($routeStruct['route_controller'])) {
        $routeStruct['route_controller'] = AdminTemplateRoute::class;
    }
}
$this['admin']['routes'] = $routes;
```

The solution would probably require customizing `AdminModule` and probably `RouteManager` to support customizable default controllers (for template, action, and script). This should be reserved for another pull request (either to be merged before or after this pull request).